packages/xdl/src/start/ExpoUpdatesManifestHandler.ts

```diff
@@ -29,14 +36,80 @@ function getPlatformFromRequest(req: express.Request | http.IncomingMessage): st
   return String(platform);
 }
 
+async function shouldUseAnonymousManifestAsync(): Promise<boolean> {
+  const currentSession = await UserManager.getSessionAsync();
+  return !currentSession || ConnectionStatus.isOffline();
+}
+
+async function getScopeKeyAndProjectIdForExpoConfigAsync(
+  expoConfig: ExpoConfig
+): Promise<{
+  scopeKey: string;
+  projectId: string | null;
+}> {
+  const shouldUseAnonymousManifest = await shouldUseAnonymousManifestAsync();
+  if (shouldUseAnonymousManifest) {
+    const userAnonymousIdentifier = await UserSettings.getAnonymousIdentifierAsync();
+    return {
+      scopeKey: `@${ANONYMOUS_USERNAME}/${expoConfig.slug}-${userAnonymousIdentifier}`,
+      projectId: null,
+    };
+  }
+
+  const user = await UserManager.ensureLoggedInAsync();
+  const username = await UserManager.getCurrentUsernameAsync();
+  const accountName = expoConfig.owner ?? username;
+  const projectName = expoConfig.slug;
+
+  try {
+    const projects = await ApiV2.clientForUser(user).getAsync('projects', {
+      experienceName: `@${accountName}/${projectName}`,
+    });
+    const project = projects[0];
+    return {
+      scopeKey: project.scopeKey,
+      projectId: project.id,
+    };
+  } catch (e) {
+    if (!(e instanceof ApiV2Error)) {
+      throw e;
+    }
+
+    if (e.code !== 'EXPERIENCE_NOT_FOUND') {
+      throw e;
+    }
+  }
+
+  const { id, scopeKey } = await ApiV2.clientForUser(user).postAsync('projects', {
```

Comment:
Should this be GET instead of POST? Also surprised this doesn't return an array like getAsync('projects') does.